### PR TITLE
ROM import by system

### PR DESCRIPTION
**Pull request by blathers16**
I think I took care of most of your concerns.  A potential future improvement is to generate the import by system menu using a Python script before opening it (I don't know enough about shell scripting to begin to attempt this with a bash script, but I am pretty confident in Python).  For now, though, I think this does what it should, let me know if you have any questions or things you want changed.

### CODE REVIEW

File: config/retronas.json.orig
Comment (sairuk) [inline config/retronas.json.orig:1]: This file shouldn't exist in the commit history

File: dialog/romimportsystem.sh [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    case ${CHOICE} in
+      01)
+        EXIT_OK
+      ;;
+      0[2-9]|[1-2][0-9]|30)
+        CLEAR
+        /opt/retronas/scripts/romimport.sh -t ${SYSTEM_ARR[10#$CHOICE]}
+        PAUSE
+      ;;
+      # 03)
```
Comment (sairuk) [inline dialog/romimportsystem.sh:49]: These comments should be deleted if your code is functional

Comment (blathers16) [reply dialog/romimportsystem.sh]: sorry, just pushed an update for that, forgot to commit.

File: config/retronas.json.orig
Comment (blathers16) [reply config/retronas.json.orig:1]: Do you want me to actually remove it from the commit history?

File: config/menu/romimport.json [hunk 2]
```diff
@@ -8,24 +8,24 @@
...
+                "index":"01",
+                "title": "Import All",
+                "description": "Run for all systems",
+                "id":"romimport-all",
+                "prompt": "",
                 "type": "script",
                 "group":"",
                 "command": "romimport"
-            },
+            },
```
Comment (sairuk) [inline config/menu/romimport.json:19]: unneeded whitespace

File: dialog/romimportsystem.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  while true
+  do
+    source $_CONFIG
+    
+    case ${CHOICE} in
+      01)
+        EXIT_OK
+      ;;
+      0[2-9]|[1-2][0-9]|30)
```
Comment (sairuk) [inline dialog/romimportsystem.sh:44]: I think this can be simplified to 

`[!01] | [0-9][0-9])`

**Comment by sairuk** [comment]
Once these last few things are addressed and tested i think i'm happy enough with this. as mentioned I think we'll need to do something about the intro screen but that is not a huge deal for this pr

File: dialog/romimportsystem.sh (hunk 3 above)
Comment (blathers16) [reply dialog/romimportsystem.sh:44]: `[!01]`
Doesn't do what you think it does.
My understanding is that that syntax matches either ! or 0 or 1, as it is a character group construct borrowed from regular expressions.

Comment (sairuk) [reply dialog/romimportsystem.sh:44]: yes you are correct, it should be `! 01`

**Comment by blathers16** [comment]
Hi @sairuk, I removed the whitespace.  I don't think the other change you requested will work, I suggested an alternative, or we could leave it as is.

File: dialog/romimportsystem.sh (hunk 3 above)
Comment (sairuk) [reply dialog/romimportsystem.sh:44]: arguably tho i think its first match anyway so we'd always match on 01 and break.

Comment (blathers16) [reply dialog/romimportsystem.sh]: I'm not sure `!01` will work either, but I think fallthrough is probably the best solution in this case anyways.

**Comment by blathers16** [comment]
hi @sairuk, I updated that last section to use fallthrough, and the fact that that case cannot be reached if $CHOICE is 01.

**Comment by sairuk** [comment]
Thanks for the submission, this has now be merged

**Pull request merged**